public/components/DataSourceThreatAlertsCard/DataSourceThreatAlertsCard.tsx

```diff
@@ -46,7 +46,13 @@ export const DataSourceThreatAlertsCard: React.FC<DataSourceAlertsCardProps> = (
   getDataSourceMenu,
   detectorService,
 }) => {
-  const DataSourceSelector = getDataSourceMenu?.();
+  const DataSourceSelector = useMemo(() => {
+    if (getDataSourceMenu) {
+      return getDataSourceMenu();
+    }
+
+    return null;
+  }, [getDataSourceMenu]);
   const notifications = getNotifications();
   const [loading, setLoading] = useState(false);
   const [dataSource, setDataSource] = useState<DataSourceOption>({
```

Comment:
do we want to remove the initial value for dataSource, that will resolve the toast
![image](https://github.com/user-attachments/assets/4b54897f-9a5f-4cca-9442-e9b9a5cc1629)

Reply:
The initial value is useful when hideLocalCluster is disabled. For better experience, we should detect the feature flag `hideLocalCluster` to give different default value for this state. I'd suggest we raise another PR to address that.